### DISCUSSION TITLE
How to track using

### DISCUSSION THREAD

**Discussion by SmithaRajan777** [add2cal/add-to-calendar-button#416] [Q&A] [answered]
Hello

How do i track which option was clicked by the user, for example, google, ical, etc.? 

I want to track that and add it to my stats so that I know which option was selected by the user

`            var element = document.querySelector('add-to-calendar-button');
            setTimeout(function() {
            element.setAttribute('data-temptext', 'whatever');
            }, 5000)

            var observer = new MutationObserver(function(mutations) {
            mutations.forEach(function(mutation) {

              

                console.log('INLOOP old:', mutation.oldValue);
                console.log('INLOOP new:', mutation.target.getAttribute("atcb-last-event"))      ;
                console.log('INLOOP new buttonId:', mutation.target.getAttribute("buttonID"))      ;

                if (mutation.type === "attributes") {
                console.log("attributes changed");

                // Example of accessing the element for which 
                // event was triggered
                //mutation.target.textContent = "Attribute of the element changed";
                console.log ("Attribute of the element changed");

                console.log('old:', mutation.oldValue);
                console.log('new:', mutation.target.getAttribute("atcb-last-event"))      ;
                }
                
                console.log(mutation.target);
            });
            });

            observer.observe(element, {
            attributes: true //configure it to listen to attribute changes
            });
            
`

I can see atcb-btn-1, for example

Thanks in advance

**Comment by jekuer** [comment, marked as answer]
What do you want to do with this information?
You could, for example, simply use the dataLayer, which gives you the following information:
<img width="561" alt="dl" src="https://github.com/add2cal/add-to-calendar-button/assets/8572883/8a17b29d-d4ed-45bd-be20-588347b1bb04">

In case you are using Google tools, it is all prepared already.

If you want to observer the changing attribute, try something like this:


```
<add-to-calendar-button
  id="myButton"
  (...)
></add-to-calendar-button

// ---

const element = document.getElementById('myButton');
var observer = new MutationObserver(function(mutations) {
  mutations.forEach(function(mutation) {
    if (mutation.type === "attributes" && mutation.attributeName === 'atcb-last-event') {
      console.log(mutation.target.getAttribute('atcb-last-event'));
    }
  });
});
observer.observe(element, {
  attributes: true
});
```
